packages/contracts/src/execute/depositCollateral.ts

```diff
@@ -25,8 +25,17 @@ export async function depositCollateral({
   if (referralCode) {
     return endpoint[
       'depositCollateralWithReferral(bytes12,uint32,uint128,string)'
-    ](bytesSubaccountName, productId, amount, referralCode);
+    ](
+      bytesSubaccountName,
+      productId,
+      toBigDecimal(amount).toFixed(0),
+      referralCode,
+    );
   } else {
-    return endpoint.depositCollateral(bytesSubaccountName, productId, amount);
+    return endpoint.depositCollateral(
+      bytesSubaccountName,
+      productId,
+      toBigDecimal(amount).toFixed(0),
```

Comment:
So casting all to `string` instead of previously `bigint`. Because it seems like it can't be serialized.

Reply:
Hmm that's very weird, this is going through ethers, where `BigNumberish` includes `bigint`. 

I don't see any related issues: https://github.com/ethers-io/ethers.js/issues?q=is%3Aissue%20Do%20not%20know%20how%20to%20serialize%20a%20BigInt 

Perhaps this is worth opening an issue with them?

Reply:
Also might be worth extracting a few utils (ex. `toBigInt(val: BigDecimalish)`, `toIntegerString(val: BigDecimalish)`)

Reply:
So my bad this can stay `bigint` and use `toBigInt` as we are calling `ethers` directly.

I realized there is a [comment](https://github.com/vertex-protocol/vertex-sdk/blob/9a3b1655b39ae1596d00f692ece97d540f156b5a/packages/contracts/src/eip712/getVertexEIP712Values.ts#L40) about it for `getVertexEIP712Values` where we have json stringiny so this has to use `toNumberString`.